modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -75,23 +75,9 @@ module ChapelIteratorSupport {
     return i.type;
   }
 
-  // Returns the type of the array that is created
-  // when this iterator is promoted to an array.
-  pragma "no doc"
-  proc iteratorToArrayType(type t) type {
-    return t;
-  }
-  pragma "no doc"
-  proc iteratorToArrayType(type t:_iteratorRecord) type {
-    var A:[1..0] iteratorToArrayType(__primitive("scalar promotion type", t));
-    return A.type;
-  }
-  pragma "no doc"
   proc iteratorToArrayElementType(type t:_iteratorRecord) type {
     return chpl__unref(
```

Comment:
is the chpl__unref call still necessary here? I have a feeling it's no longer needed. Could you check?

Reply:
The only test that fails with chpl__unref() removed is:

    expressions/loop-expr/zip-arrays.chpl

I am leaving the investigation as future work.

Reply:
Could you leave a comment here next to the chpl__unref call to point out that it might not be necessary, but one should check that test first?